Lib/Protocols/IdCompressorZLib.pas

```diff
@@ -172,7 +172,11 @@ procedure TIdCompressorZLib.InternalDecompressStream(
     repeat
       LZstream.next_out := @outBuffer[0];
       LZstream.avail_out := bufferSize;
-      DCheck(inflate(LZstream,Z_NO_FLUSH));
+      zresult := inflate(LZstream, Z_NO_FLUSH);
+      if zresult <> Z_BUF_ERROR then

```

Comment:
I understand that `Z_BUF_ERROR` is not a fatal error from `inflate`'s POV (there is a big comment in the code stating as much), but in this context isn't it really a fatal error? It is being given a 32K output buffer.  If "no progress was possible" at all with that much space provided to it, then how is ignoring `Z_BUF_ERROR` supposed to allow it to make progress?  Are you suggesting that it is still progressing through the input buffer, just not writing anything at all to the output buffer?

Reply:
Delphi do what I did (System.ZLib.pas):
function ZDecompressCheckWithoutBufferError(code: Integer): Integer; overload;
begin
  Result := code;

  if code < 0 then
    if (code <> Z_BUF_ERROR) then
     raise EZDecompressionError.Create(string(_z_errmsg[2 - code])) at ReturnAddress;
end;